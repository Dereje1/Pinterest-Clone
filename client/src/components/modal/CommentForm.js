import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';

function CommentForm({ handleClose, handleSubmit }) {
  const [comment, setComment] = useState('');

  return (
    <div style={{
      margin: 5, display: 'flex', flexDirection: 'column', alignItems: 'flex-end',
    }}
    >
      <TextField
        autoFocus
        id="comment_form"
        label="Add a comment"
        type="text"
        multiline
        rows={3}
        fullWidth
        value={comment}
        onChange={(e) => setComment(e.target.value)}
      />
      <div style={{ marginTop: 1 }}>
        <Button onClick={handleClose}>Cancel</Button>
        <Button onClick={() => handleSubmit(comment)}>Submit</Button>
      </div>
    </div>
  );
}

export default CommentForm;

CommentForm.propTypes = {
  handleClose: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};
