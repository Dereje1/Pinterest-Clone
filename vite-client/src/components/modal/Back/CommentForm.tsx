import React, { useState } from 'react';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';

interface CommentFormProps {
  handleClose: () => void
  handleSubmit: (comment: string) => void
}

function CommentForm({ handleClose, handleSubmit }: CommentFormProps) {
  const [comment, setComment] = useState('');

  return (
    <div style={{
      margin: 5, display: 'flex', flexDirection: 'column', alignItems: 'flex-end',
    }}
    >
      <TextField
        autoFocus
        id="comment_form"
        label="Add a comment"
        type="text"
        multiline
        rows={3}
        fullWidth
        value={comment}
        onChange={(e) => setComment(e.target.value)}
        sx={{ caretColor: 'black', marginTop: 2 }}
      />
      <div style={{ marginTop: 1 }}>
        <Button onClick={handleClose}>Cancel</Button>
        <Button onClick={() => handleSubmit(comment)}>Submit</Button>
      </div>
    </div>
  );
}

export default CommentForm;
