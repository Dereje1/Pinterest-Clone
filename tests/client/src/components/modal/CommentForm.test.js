import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommentForm from '../../../../../client/src/components/modal/CommentForm';

test('will render the form to add comments', () => {
  const wrapper = shallow(
    <CommentForm
      open
      handleClose={jest.fn()}
      handleSubmit={jest.fn()}
    />,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('will update the comment in state (controlled)', () => {
  const wrapper = shallow(
    <CommentForm
      open
      handleClose={jest.fn()}
      handleSubmit={jest.fn()}
    />,
  );
  let textField = wrapper.find('ForwardRef(TextField)');
  expect(textField.props().value).toBe('');
  textField.props().onChange({ target: { value: 'new comment' } });
  textField = wrapper.find('ForwardRef(TextField)');
  expect(textField.props().value).toBe('new comment');
});

test('will submit a comment', () => {
  const handleSubmit = jest.fn();
  const wrapper = shallow(
    <CommentForm
      open
      handleClose={jest.fn()}
      handleSubmit={handleSubmit}
    />,
  );
  const textField = wrapper.find('ForwardRef(TextField)');
  expect(textField.props().value).toBe('');
  textField.props().onChange({ target: { value: 'new comment to submit' } });
  const submit = wrapper.find('ForwardRef(Button)').at(1);
  submit.props().onClick();
  expect(handleSubmit).toHaveBeenCalledWith('new comment to submit');
});
