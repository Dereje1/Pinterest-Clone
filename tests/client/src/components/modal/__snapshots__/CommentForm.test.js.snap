// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render the form to add comments 1`] = `
<div>
  <ForwardRef(Dialog)
    fullWidth={true}
    onClose={[MockFunction]}
    open={true}
  >
    <ForwardRef(DialogContent)>
      <ForwardRef(TextField)
        autoFocus={true}
        fullWidth={true}
        id="comment_form"
        label="Add a comment"
        margin="dense"
        multiline={true}
        onChange={[Function]}
        rows={3}
        type="text"
        value=""
      />
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        onClick={[MockFunction]}
      >
        Cancel
      </ForwardRef(Button)>
      <ForwardRef(Button)
        onClick={[Function]}
      >
        Submit
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</div>
`;
